src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettings.cs

```diff
@@ -157,6 +163,75 @@ public async Task SetInstalledTemplatePackagesAsync(IReadOnlyList<TemplatePackag
             throw new InvalidOperationException();
         }
 
+        public async Task MigrateCacheMetadataAsync(CancellationToken cancellationToken)
```

Comment:
whole method should be locked as here https://github.com/dotnet/templating/blob/6536e8d9b4ae1e70047cf4c254c352db75b99903/src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettingsTemplatePackageProvider.cs#L171
before cookie is checked, so migration is not performed twice.